Java-Control-Statements.md

```diff
@@ -0,0 +1,216 @@
+`.
+Java has two selection statements : `if-else` and `switch`.
+
+###if-else
+
+`if-else` is a conditional branch statement that helps us to choose between two available choices. The general syntax of `if-else` is
+shown below :
+
+```
+if(condition)
+statement1;
+else
+statement2;
+```
+
+In the above code snippet, the `condition` is first resolved. If it turns out to be true, then the statements that are inside the if 
+condition are executed(`statement1 in this case`). If the `condition` turns out to be false, then the statements inside else are 
+executed(`statement2, in this case`). 
+
+**Example :**
+
+Here is an example usage which states whether a number is an `even number or an odd number`. Even numbers are divisible by two
+and hence leave a remainder of zero. The remainder on dividing the number by two can be obtained using the `modulo (%) mathematical
+operator`. 
+
+```
+public class Test {
+   public static void main(String args[])
+      {
+   int num=5;
+   
+if ( num%2==0){ 
+    System.out.println("Number is even");
+}
+else{
+    System.out.println("Number is odd");
+       }
+   }
+}
+```
+
+**Output : Number is odd**
+
+####Nested if-else
+
+`Nested if-else` is very common in programming. Here one if statement is a target of another or subsequent `if-else statements`.
+
+**Example :**
+
+```
+public class Test {
+   public static void main(String args[])
+      {
+   
+   int num=5;
+
+if (num == 1) {
+    System.out.println("One");     //first if statement
+} else {
+    if (num == 2) {
+        System.out.println("Two");
+    } else {
+        if (num == 3) {
+            System.out.println("Three");      //second if statement
+        } else {
+            System.out.println("Number is greater than three");
+        }
+    }
+}
+```
+
+**Output : Number is greater than three**
+
+####The if-else-if ladder
+
+The above code is highly unreadable. To make it better we can use chained `else-if statements` with the first `if statement`. 
+
+```
+public class Test {
+   public static void main(String args[])
+       {
+   int num=3
+
+if (num == 1) {
+    System.out.println("One");
+} else if (num == 2) {
+    System.out.println("Two");
+} else if (num == 3) {
+    System.out.println("Three");
+} else {
+    System.out.println("Numbers greater than three cannot be processed by this code");
+}
+```
+
+**Output : Three**
+
+Here the if statement is executed from `top-down`. As soon as one of the `if condition` is true the statements associated with it are
+executed, and the rest of the ladder is bypassed. If none of the conditions are true then the final `else statement` is executed as a
+`default condition`.
+
+###switch
+
+The `switch` staement is Java's multiway branching statement. It provides an easy way to dispatch execution to a desired path based on
+the value of an expression. It is actually  a better alternative than using large `if-else statements`.
```

Comment:
It's not always. If statements can compare between two variables. Switch can compare only between a variable and several possible **constant** values.